### PR TITLE
Network layout algorithm

### PR DESCRIPTION
**Pull request by Vortrox**
This is a custom network layout algorithm that calculates where each unit should be positioned initially. It uses Tensorflow's implementation of the backpropagation algorithm to achieve this, which is easy to parallelize if necessary.

### CODE REVIEW

**Comment by Vortrox** [comment]
This is the current result of this custom network layout algorithm.
image
image

It works, but there are a few major problems with it:
- The algorithm doesn't properly minimize the distance of visible edges, so we get edges going across the entire diagram
- Clusters tend to form in a line along the diagonals instead of spreading out across the entire diagram
- There are a lot of edges crossing over each other, causing the layout within clusters to be unintuitive
- There's an unresolved bug that's stopping the algorithm from working with batch sizes greater than 1, meaning it can't make full use of Tensorflow's batch processing parallelization
Controlling the layout is very difficult without proper intuition about how force-based network layout algorithms and its underlying math works. At this point I am considering scrapping this algorithm in favor of another approach.

**Comment by Vortrox** [comment]
I'm discontinuing development of this algorithm in favor of the one in #31 due to the complexity of the mathematics needed to describe a loss function which results in a good network layout.

**Comment by Vortrox** [comment]
[Custom network layout algorithm based on neural network SGD algorithm](https://trello.com/c/6iKJSuBW/179-custom-network-layout-algorithm-based-on-neural-network-sgd-algorithm)
